modules/op-about-pipelinerun.adoc

```diff
@@ -3,13 +3,13 @@
 // *openshift_pipelines/op-creating-applications-with-cicd-pipelines.adoc
 
 [id="about-pipelinerun_{context}"]
-= PipelineRun
+= Pipeline run
```

Comment:
"PipelineRun" is the name of a Kubernetes resource. So let us keep it as it is. Yes, this type functionally represents a running instance of a "Pipeline". Stil the resource we are talking about here is of type `PipelineRun`. 

In other words, a Pipeline resource is an abstract representation of a set of things (Tasks) we need to accomplish as part of a CICD workflow. We create a resource of type PipelineRun which binds a pipeline, workspaces, credentials and a set of parameter values specific to a scenario to execute the CICD workflow.

In short, a developer will expect to see the word PipelineRun (in camel case) not pipeline run. I think moving away from these expectations will create confusion. (Especially there is a new "Run" type, different from Pipeline, PipelineRun, Task, and TaskRun in upstream, which we will have to support few releases down the road) 🧑‍💻

Reply:
@sounix000 Please review the following guidelines for object references, here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references. Perhaps we could retitle this to:
```suggestion
= About PipelineRun resources
```

Reply:
@nikhil-thomas @rolfedh - Rant below, but please bear with me.

It was many months back that I had a detailed discussion with @Preeticp about using CamelCase vs sentence case for such things. We arrived at the conclusion that wherever possible, we will use sentence case, i.e., we will use the corresponding actions. The reason is primarily one of our style guide conventions, which mandates mentioning a type immediately after objects. For example, I have to write "`PipelineRun` resource" every single time I mention PipelineRun. In a typical Pipelines document, we mention PipelineRun may be 10 times or more, and it would be a nightmare for the reader to read the type after each mention.

I found such a convention a bit stifling, and looked for a workaround. Personally speaking, I am not very fond of converting objects/resources to their corresponding actions. I understand that developers are accustomed to a certain terminology, and there are certain implicit expectations. But I am also not comfortable mentioning the type every single time. As a result, over the last 8 months, wherever possible I have been converting all CamelCase to sentence case. Until recently, we even had a JIRA task to track the effort.

At the moment, if I change this particular instance to CamelCase, it will be a violation of consistency. Also, if I try to maintain consistency, it will mean that I should change sentence case to CamelCase throughout pipelines documentation, it's going to be a huge and monotonous effort.

I'm in a real tight spot here. @Preeticp please weigh in your opinion.

Reply:
I hear you, @sounix000. I trust you'll evaluate the different perspectives and make a sound decision. My role here is to make suggestions, not decrees. 

Initially, reading Nikhil's statement, I was thinking the same as you: "Sorry - we have discussed and chosen a naming convention and must stick to it." 

However, (and this is open for discussion) I think what Nikhil is saying is that "pipeline run" is literally not the same thing as a "PipelineRun" -- In other words that a running pipeline is not the same thing as a PipelineRun resource. And that our documentation must make this distinction clear. Please correct me if I'm wrong. (As a reviewer, I don't have the time to learn the product in depth.)

As for making tedious repetitive references to the thing, whatever it is called, one solution is to call the first instance in a paragraph by its full name, "PipelineRun resource," and then refer to subsequent instances as "the resource" unless the context requires further clarification (such as when you're discussing multiple types of resources.)